### DISCUSSION TITLE
Is it OK that `AbstractBuiltResponse` effectively ignores provided `reasonPhrase`?

### DISCUSSION THREAD

**Discussion by jakubzytka** [resteasy/resteasy#2947] [Q&A]
Please bear with me if this was clarified gazillion times already...

Consider the following (minimized for brevity):
```
response = Response.status(500, "bulbulator ran out of water").build();

assertEquals("bulbul....", response.getStatusInfo().getReasonPhrase()); <-- fails; the reason phrase is "Internal Server Error" 
```

this is due to:
https://github.com/resteasy/resteasy/blob/1cca43bb27e323873610754958f6f3ad46f0df9d/resteasy-core/src/main/java/org/jboss/resteasy/specimpl/AbstractBuiltResponse.java#L132-L160

I.e. the provided `reasonPhrase` is ignored when a status code is defined as a "commonly used" one by `javax.ws.rs.core.Response.Status`.

Is it OK? It is surprising and the code itself looks like conflicting decisions had been made:
- on one hand, the API supports providing a reason phrase
- on the other, `AbstractBuiltResponse` is happy to  ignore it
- there is a `public String AbstractBuiltResponse::getReasonPhrase()` method, but the reason phrase is really a property of the status info (`StatusType`)
- finally, `AbstractBuiltResponse` ignores the reason phrase when a `Status` with the same status code is defined in an enum `Status`. This enum looks more like an entity for convenience (it's a non-exhaustive list of common statuses), and which, potentially, can be extended at any time, breaking backward compatibility (reason phrase disappears for some newly added enum value)

**Comment by jamezp** [comment]
This seems a bit odd to me as well. It looks like it was introduced in [RESTEASY-1991](https://issues.redhat.com/browse/RESTEASY-1991), https://github.com/resteasy/resteasy/pull/1835. @rsearls this was a while ago, but do you recall if there was a reason for this?

**Reply by jamezp** [reply]
Something like this seems more correct https://github.com/jamezp/resteasy/commit/7c59d68aa3917d7b14b07a06840d0dcf05e2574b.
